src/rgw/driver/d4n/d4n_policy.h

```diff
@@ -1,104 +1,138 @@
-#ifndef CEPH_D4NPOLICY_H
-#define CEPH_D4NPOLICY_H
+#pragma once
 
 #include <string>
 #include <iostream>
-#include <cpp_redis/cpp_redis>
 #include "rgw_common.h"
 #include "d4n_directory.h"
 #include "../../rgw_redis_driver.h"
 
+#define dout_subsys ceph_subsys_rgw
+#define dout_context g_ceph_context
+
 namespace rgw { namespace d4n {
 
 class CachePolicy {
-  private:
-    cpp_redis::client client;
-    Address addr;
-
-  public:
-    CephContext* cct;
+  protected:
+    struct Entry : public boost::intrusive::list_base_hook<> {
+      std::string key;
+      uint64_t offset;
+      uint64_t len;
+      std::string version;
+      Entry(std::string& key, uint64_t offset, uint64_t len, std::string version) : key(key), offset(offset), 
+                                                                                     len(len), version(version) {}
+    };
+    
+    //The disposer object function
+    struct Entry_delete_disposer {
+      void operator()(Entry *e) {
+        delete e;
+      }
+    };
+    typedef boost::intrusive::list<Entry> List;
 
-    CachePolicy() : addr() {}
-    virtual ~CachePolicy() = default;
+    std::unordered_map<std::string, Entry*> entries_map;
```

Comment:
make this private to the LRU Policy class since `entries_map`, will only be used by LRU Policy class now